### ISSUE TITLE
Can't build zfs-0.8.3 module with 5.6.6 kernel on Fedora

### ISSUE THREAD

**Issue by olegon-ru** [openzfs/zfs#10257]
It's impossible to build zfs module on Fedora Workstation with kernel 5.6.6 (confirmed on a few PC)

dkms module installation failed with errors... Suppose it's kernel incompatibility, because it's works well after kernel downgrade...

https://paste.centos.org/view/raw/dc740dad
```
DKMS make.log for zfs-0.8.3 for kernel 5.6.6-200.fc31.x86_64 (x86_64)
Sun Apr 26 12:09:39 MSK 2020
./scripts/make_gitrev.sh
make: [Makefile:1544: gitrev] Error 128 (ignored)
make  all-recursive
make[1]: Entering directory '/var/lib/dkms/zfs/0.8.3/build'
Making all in include
make[2]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include'
Making all in linux
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/linux'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/linux'
Making all in spl
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/spl'
Making all in rpc
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/spl/rpc'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/spl/rpc'
Making all in sys
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/spl/sys'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/spl/sys'
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/spl'
make[4]: Nothing to be done for 'all-am'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/spl'
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/spl'
Making all in sys
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys'
Making all in fm
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fm'
Making all in fs
make[5]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fm/fs'
make[5]: Nothing to be done for 'all'.
make[5]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fm/fs'
make[5]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fm'
make[5]: Nothing to be done for 'all-am'.
make[5]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fm'
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fm'
Making all in fs
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fs'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/fs'
Making all in crypto
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/crypto'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/crypto'
Making all in lua
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/lua'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/lua'
Making all in sysevent
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/sysevent'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys/sysevent'
make[4]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include/sys'
make[4]: Nothing to be done for 'all-am'.
make[4]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys'
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include/sys'
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/include'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include'
make[2]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/include'
Making all in rpm
make[2]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/rpm'
Making all in generic
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/rpm/generic'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/rpm/generic'
Making all in redhat
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/rpm/redhat'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/rpm/redhat'
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/rpm'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/rpm'
make[2]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/rpm'
Making all in module
make[2]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/module'
list='icp lua'; for targetdir in $list; do \
	make -C $targetdir; \
done
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/module/icp'
mkdir -p api core spi io os algs algs/aes algs/edonr algs/modes algs/sha1 algs/sha2 algs/skein asm-x86_64 asm-x86_64/aes asm-x86_64/modes asm-x86_64/sha1 asm-x86_64/sha2 asm-i386 asm-generic
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/module/icp'
make[3]: Entering directory '/var/lib/dkms/zfs/0.8.3/build/module/lua'
mkdir -p setjmp
make[3]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/module/lua'
make -C /lib/modules/5.6.6-200.fc31.x86_64/build M=`pwd`  CONFIG_ZFS=m modules
make[3]: Entering directory '/usr/src/kernels/5.6.6-200.fc31.x86_64'
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/lua/lapi.o
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/icp/illumos-crypto.o
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/avl/avl.o
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/nvpair/nvpair.o
In file included from /var/lib/dkms/zfs/0.8.3/build/include/sys/nvpair.h:30,
                 from /var/lib/dkms/zfs/0.8.3/build/module/nvpair/nvpair.c:30:
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:88:15: error: unknown type name 'time_t'
   88 | static inline time_t
      |               ^~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h: In function 'gethrtime':
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: error: storage size of 'ts' isn't known
  108 |  struct timespec ts;
      |                  ^~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:109:2: error: implicit declaration of function 'getrawmonotonic' [-Werror=implicit-function-declaration]
  109 |  getrawmonotonic(&ts);
      |  ^~~~~~~~~~~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: warning: unused variable 'ts' [-Wunused-variable]
  108 |  struct timespec ts;
      |                  ^~
In file included from /var/lib/dkms/zfs/0.8.3/build/include/spl/sys/condvar.h:33,
                 from /var/lib/dkms/zfs/0.8.3/build/include/sys/zfs_context.h:38,
                 from /var/lib/dkms/zfs/0.8.3/build/include/sys/crypto/common.h:39,
                 from /var/lib/dkms/zfs/0.8.3/build/module/icp/illumos-crypto.c:35:
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:88:15: error: unknown type name 'time_t'
   88 | static inline time_t
      |               ^~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h: In function 'gethrtime':
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: error: storage size of 'ts' isn't known
  108 |  struct timespec ts;
      |                  ^~
In file included from /var/lib/dkms/zfs/0.8.3/build/include/spl/sys/condvar.h:33,
                 from /var/lib/dkms/zfs/0.8.3/build/include/sys/zfs_context.h:38,
                 from /var/lib/dkms/zfs/0.8.3/build/include/sys/lua/lua.h:13,
                 from /var/lib/dkms/zfs/0.8.3/build/module/lua/lapi.c:12:
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:88:15: error: unknown type name 'time_t'
   88 | static inline time_t
      |               ^~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:109:2: error: implicit declaration of function 'getrawmonotonic' [-Werror=implicit-function-declaration]
  109 |  getrawmonotonic(&ts);
      |  ^~~~~~~~~~~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: warning: unused variable 'ts' [-Wunused-variable]
  108 |  struct timespec ts;
      |                  ^~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h: In function 'gethrtime':
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: error: storage size of 'ts' isn't known
  108 |  struct timespec ts;
      |                  ^~
  LD [M]  /var/lib/dkms/zfs/0.8.3/build/module/avl/zavl.o
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:109:2: error: implicit declaration of function 'getrawmonotonic' [-Werror=implicit-function-declaration]
  109 |  getrawmonotonic(&ts);
      |  ^~~~~~~~~~~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: warning: unused variable 'ts' [-Wunused-variable]
  108 |  struct timespec ts;
      |                  ^~
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/spl/spl-atomic.o
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:268: /var/lib/dkms/zfs/0.8.3/build/module/icp/illumos-crypto.o] Error 1
make[4]: *** [scripts/Makefile.build:505: /var/lib/dkms/zfs/0.8.3/build/module/icp] Error 2
make[4]: *** Waiting for unfinished jobs....
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/nvpair/fnvpair.o
cc1: some warnings being treated as errors
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:268: /var/lib/dkms/zfs/0.8.3/build/module/nvpair/nvpair.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[5]: *** [scripts/Makefile.build:268: /var/lib/dkms/zfs/0.8.3/build/module/lua/lapi.o] Error 1
make[5]: *** Waiting for unfinished jobs....
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/lua/lauxlib.o
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/spl/spl-condvar.o
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/spl/spl-cred.o
In file included from /var/lib/dkms/zfs/0.8.3/build/include/sys/nvpair.h:30,
                 from /var/lib/dkms/zfs/0.8.3/build/module/nvpair/fnvpair.c:26:
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:88:15: error: unknown type name 'time_t'
   88 | static inline time_t
      |               ^~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h: In function 'gethrtime':
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: error: storage size of 'ts' isn't known
  108 |  struct timespec ts;
      |                  ^~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:109:2: error: implicit declaration of function 'getrawmonotonic' [-Werror=implicit-function-declaration]
  109 |  getrawmonotonic(&ts);
      |  ^~~~~~~~~~~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: warning: unused variable 'ts' [-Wunused-variable]
  108 |  struct timespec ts;
      |                  ^~
In file included from /var/lib/dkms/zfs/0.8.3/build/include/spl/sys/condvar.h:33,
                 from /var/lib/dkms/zfs/0.8.3/build/module/spl/spl-condvar.c:27:
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:88:15: error: unknown type name 'time_t'
   88 | static inline time_t
      |               ^~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h: In function 'gethrtime':
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: error: storage size of 'ts' isn't known
  108 |  struct timespec ts;
      |                  ^~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:268: /var/lib/dkms/zfs/0.8.3/build/module/nvpair/fnvpair.o] Error 1
make[4]: *** [scripts/Makefile.build:505: /var/lib/dkms/zfs/0.8.3/build/module/nvpair] Error 2
  CC [M]  /var/lib/dkms/zfs/0.8.3/build/module/spl/spl-err.o
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:109:2: error: implicit declaration of function 'getrawmonotonic' [-Werror=implicit-function-declaration]
  109 |  getrawmonotonic(&ts);
      |  ^~~~~~~~~~~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: warning: unused variable 'ts' [-Wunused-variable]
  108 |  struct timespec ts;
      |                  ^~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:111:1: warning: control reaches end of non-void function [-Wreturn-type]
  111 | }
      | ^
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:268: /var/lib/dkms/zfs/0.8.3/build/module/spl/spl-condvar.o] Error 1
make[5]: *** Waiting for unfinished jobs....
In file included from /var/lib/dkms/zfs/0.8.3/build/include/spl/sys/condvar.h:33,
                 from /var/lib/dkms/zfs/0.8.3/build/include/sys/zfs_context.h:38,
                 from /var/lib/dkms/zfs/0.8.3/build/include/sys/lua/lua.h:13,
                 from /var/lib/dkms/zfs/0.8.3/build/module/lua/lauxlib.c:16:
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:88:15: error: unknown type name 'time_t'
   88 | static inline time_t
      |               ^~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h: In function 'gethrtime':
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: error: storage size of 'ts' isn't known
  108 |  struct timespec ts;
      |                  ^~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:109:2: error: implicit declaration of function 'getrawmonotonic' [-Werror=implicit-function-declaration]
  109 |  getrawmonotonic(&ts);
      |  ^~~~~~~~~~~~~~~
/var/lib/dkms/zfs/0.8.3/build/include/spl/sys/time.h:108:18: warning: unused variable 'ts' [-Wunused-variable]
  108 |  struct timespec ts;
      |                  ^~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:268: /var/lib/dkms/zfs/0.8.3/build/module/lua/lauxlib.o] Error 1
make[4]: *** [scripts/Makefile.build:505: /var/lib/dkms/zfs/0.8.3/build/module/lua] Error 2
make[4]: *** [scripts/Makefile.build:505: /var/lib/dkms/zfs/0.8.3/build/module/spl] Error 2
make[3]: *** [Makefile:1683: /var/lib/dkms/zfs/0.8.3/build/module] Error 2
make[3]: Leaving directory '/usr/src/kernels/5.6.6-200.fc31.x86_64'
make[2]: *** [Makefile:30: modules] Error 2
make[2]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build/module'
make[1]: *** [Makefile:839: all-recursive] Error 1
make[1]: Leaving directory '/var/lib/dkms/zfs/0.8.3/build'
make: *** [Makefile:708: all] Error 2```
-->

**Comment by ntphx** [comment]
Same issue on `5.6.6-200.fc31.x86_64`

What kernel did you downgrade to for it to work again?

**Comment by olegon-ru** [comment]
It was `5.5.16-200.fc31.x86_64` and worked very well... But I upgrade without testing to `5.6.6-200` and zfs module was lost... Unfortunately, Fedora not allows to dongrade to previous version of kernel, so I downgraded to `5.3.7-301.fc31.x86_64` by `dnf downgrade kernel`

**Comment by ntphx** [comment]
> It was `5.5.16-200.fc31.x86_64` and worked very well... But I upgrade without testing to `5.6.6-200` and zfs module was lost... Unfortunately, Fedora not allows to dongrade to previous version of kernel, so I downgraded to `5.3.7-301.fc31.x86_64` by `dnf downgrade kernel`

I've done the same thing and now it works again. So it looks like there's some sort of problem with `5.5.16-200.fc31.x86_64` compatibility.

**Comment by olegon-ru** [comment]
> > It was `5.5.16-200.fc31.x86_64` and worked very well... But I upgrade without testing to `5.6.6-200` and zfs module was lost... Unfortunately, Fedora not allows to dongrade to previous version of kernel, so I downgraded to `5.3.7-301.fc31.x86_64` by `dnf downgrade kernel`
> 
> I've done the same thing and now it works again. So it looks like there's some sort of problem with `5.5.16-200.fc31.x86_64` compatibility.

Excuse me, 5.6.6 compatibility?

**Comment by ntphx** [comment]
> > > It was `5.5.16-200.fc31.x86_64` and worked very well... But I upgrade without testing to `5.6.6-200` and zfs module was lost... Unfortunately, Fedora not allows to dongrade to previous version of kernel, so I downgraded to `5.3.7-301.fc31.x86_64` by `dnf downgrade kernel`
> > 
> > 
> > I've done the same thing and now it works again. So it looks like there's some sort of problem with `5.5.16-200.fc31.x86_64` compatibility.
> 
> Excuse me, `5.6.6-200` compatibility?

Yes indeed I have edited my comment!

**Comment by glbyers** [comment]
This looks to be a wider problem. I'm having the same issue described above, but not only with ZFS module. Similar issues with various time structures trying to dkms build nvidia module too.

**Comment by thulle** [comment]
Excuse me if I'm missing anything, but 0.8.3 that you're trying to compile against kernel 5.6 was released as compatible with 2.6.32-5.4:
https://github.com/openzfs/zfs/releases/tag/zfs-0.8.3

Compability with 5.5 and 5.6 is discussed here:
https://github.com/openzfs/zfs/pull/9961#issuecomment-610484852

**Comment by olegon-ru** [comment]
But 5.6.6 now is current for Fedora Wrokstation (release), so it's very widespread problem of zfs-0.8.3-1.fc31.src.rpm package from official ZoL repo...

**Comment by glbyers** [comment]
> Excuse me if I'm missing anything, but 0.8.3 that you're trying to compile against kernel 5.6 was released as compatible with 2.6.32-5.4:
> https://github.com/openzfs/zfs/releases/tag/zfs-0.8.3
> 
> Compability with 5.5 and 5.6 is discussed here:
> [#9961 (comment)](https://github.com/openzfs/zfs/pull/9961#issuecomment-610484852)

I'd say this is going to catch out a lot of people. Anyone using Fedora 31 with ZoL & the official ZoL yum repos for a start. 5.6.6 just came down in Fedora 31 stable.

**Comment by ntphx** [comment]
> Excuse me if I'm missing anything, but 0.8.3 that you're trying to compile against kernel 5.6 was released as compatible with 2.6.32-5.4:
> https://github.com/openzfs/zfs/releases/tag/zfs-0.8.3
> 
> Compability with 5.5 and 5.6 is discussed here:
> [#9961 (comment)](https://github.com/openzfs/zfs/pull/9961#issuecomment-610484852)

Yep this is a bit embarrassing. As a suggestion perhaps there could be some basic sanity check that runs when people's DKMS tries to build the modules? Now it just breaks with some cryptic messages.

If it would say "Hey, your kernel isn't supported" that would probably prevent more of these issues being raised.

**Comment by NTmatter** [comment]
Thankfully the last few kernels are retained and available in the bootloader. `5.5.16-200.fc31.x86_64` and `5.5.17-200.fc31.x86_64` should still be usable right now.

If, by chance, you need to clean out and rebuild the DKMS modules after deleting them like I did (doh!):
```
ZFS_VERSION=0.8.3

rmmod zfs
dkms status
dkms remove zfs/$ZFS_VERSION --all
dkms install zfs/$ZFS_VERSION
modprobe zfs
```

**Comment by olegon-ru** [comment]
If you set up installonly_limit=2 and perform downgrade... You'll never seen 5.5.17 again.
On my machines is only available 5.3.7 and 5.6.6, so problem is very urgent... Fedora will completely upgraded by changing version in a few days, many peoples lost access to old kernels and zfs support...

**Comment by tvtue** [comment]
+1 from here.

**Comment by mariobaldini** [comment]
+1 
The latest regular `dnf update` broken my servers running ZoL.

**Comment by joshlay** [comment]
The 5.6.x kernels underwent significant changes around time_t

I don't have a reference, but there was a patch in a git issue relating to this that I've been using without issue:
```
$ uname -r
5.6.6-300.fc32.x86_64
$ lsmod | grep zfs
zfs                  4280320  9
```

**Comment by jagauthier** [comment]
I pulled the zfs git tree in and it worked. 
I renamed the existing /usr/src/zfs-0.8.3 and then renamed the git pull to zfs-0.8.3.   Then I copied dkms.conf from the old tree to the new.
It built. It inserted, and I was able to import my pool.

**Comment by GongYiLiao** [comment]
> I pulled the zfs git tree in and it worked.
> I renamed the existing /usr/src/zfs-0.8.3 and then renamed the git pull to zfs-0.8.3. Then I copied dkms.conf from the old tree to the new.
> It built. It inserted, and I was able to import my pool.

Did you need to run any script under zfs git pull?   I tried to follow your steps but get the following error:  
```
/usr/src/zfs-0.8.3 $ sudo dkms install zfs/0.8.3 -k 5.6.6-300.fc32.x86_64

Kernel preparation unnecessary for this kernel.  Skipping...

Warning: The pre_build script is not executable.

Building module:
cleaning build area...(bad exit status: 2)
make -j4 KERNELRELEASE=5.6.6-300.fc32.x86_64...(bad exit status: 2)
Error! Bad return status for module build on kernel: 5.6.6-300.fc32.x86_64 (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.

/usr/src/zfs-0.8.3 $ cat /var/lib/dkms/zfs/0.8.3/build/make.log
DKMS make.log for zfs-0.8.3 for kernel 5.6.6-300.fc32.x86_64 (x86_64)
Tue 28 Apr 2020 08:31:18 PM CDT
make: *** No targets specified and no makefile found.  Stop.


```

**Comment by glbyers** [comment]
> I pulled the zfs git tree in and it worked.
> I renamed the existing /usr/src/zfs-0.8.3 and then renamed the git pull to zfs-0.8.3. Then I copied dkms.conf from the old tree to the new.
> It built. It inserted, and I was able to import my pool.

I'd have to recommend against this. Why not just versionlock kernel to prevent update until the issue is fixed and tested properly? You always run the risk of data loss, but more so when using code that hasn't been confirmed stable yet.

`dnf versionlock kernel\*`

**Comment by jagauthier** [comment]
> > I pulled the zfs git tree in and it worked.
> > I renamed the existing /usr/src/zfs-0.8.3 and then renamed the git pull to zfs-0.8.3. Then I copied dkms.conf from the old tree to the new.
> > It built. It inserted, and I was able to import my pool.
> 
> Did you need to run any script under zfs git pull? I tried to follow your steps but get the following error:
> 
> ```
> /usr/src/zfs-0.8.3 $ sudo dkms install zfs/0.8.3 -k 5.6.6-300.fc32.x86_64
> 
> Kernel preparation unnecessary for this kernel.  Skipping...
> 
> Warning: The pre_build script is not executable.
> 
> Building module:
> cleaning build area...(bad exit status: 2)
> make -j4 KERNELRELEASE=5.6.6-300.fc32.x86_64...(bad exit status: 2)
> Error! Bad return status for module build on kernel: 5.6.6-300.fc32.x86_64 (x86_64)
> Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
> 
> /usr/src/zfs-0.8.3 $ cat /var/lib/dkms/zfs/0.8.3/build/make.log
> DKMS make.log for zfs-0.8.3 for kernel 5.6.6-300.fc32.x86_64 (x86_64)
> Tue 28 Apr 2020 08:31:18 PM CDT
> make: *** No targets specified and no makefile found.  Stop.
> ```

Yeah, i had to run autogen.sh to generate the conduit script.

**Comment by jagauthier** [comment]
> > I pulled the zfs git tree in and it worked.
> > I renamed the existing /usr/src/zfs-0.8.3 and then renamed the git pull to zfs-0.8.3. Then I copied dkms.conf from the old tree to the new.
> > It built. It inserted, and I was able to import my pool.
> 
> I'd have to recommend against this. Why not just versionlock kernel to prevent update until the issue is fixed and tested properly? You always run the risk of data loss, but more so when using code that hasn't been confirmed stable yet.
> 
> `dnf versionlock kernel\*`

Well, i just I just installed fedora 32 and that's the kernel it came with.   I haven't used Fedora since it was called redhat, so I'm just trying to make it work.

**Comment by olegon-ru** [comment]
Fedora next upgrade, kernel 5.6.7-200, and zfs module can't compiled again, as expected.

**Comment by snowsnoot** [comment]
zfs-dkms builds ok on 5.5.16-200.fc31.x86_64 and does not build on 5.6.8-200.fc31.x86_64

**Comment by jdmaloney** [comment]
> Well, i just I just installed fedora 32 and that's the kernel it came with. I haven't used Fedora since it was called redhat, so I'm just trying to make it work.

Just for reference/future knowledge, RedHat most definitely still exists and is the “production” distro (and CentOS is the community supported free version of it), Fedora is the bleeding edge development distro for RedHat

**Comment by ElCoyote27** [comment]
Any hope to perhaps see a 0.8.4 release for Fedora soon? I'm still on FC30 but as soon as I got the 5.6 kernels, zfs-dkms could no longer be compiled.

**Comment by mskarbek** [comment]
> Just for reference/future knowledge, RedHat most definitely still exists and is the “production” distro (and CentOS is the community supported free version of it), Fedora is the bleeding edge development distro for RedHat

@jdmaloney it is leading not bleeding and more than just a development distro. ;)

It always baffles me, whit every issue like this, what is it so hard to understand in the phrase "Compatible with 2.6.32 - 5.4 Linux kernels" on the release page/in the META file?

**Comment by ntphx** [comment]
> > Just for reference/future knowledge, RedHat most definitely still exists and is the “production” distro (and CentOS is the community supported free version of it), Fedora is the bleeding edge development distro for RedHat
> 
> @jdmaloney it is leading not bleeding and more than just a development distro. ;)
> 
> It always baffles me, with every issue like this, what is it so hard to understand in the phrase "Compatible with 2.6.32 - 5.4 Linux kernels" on the release page/in the META file?

It happens because people make mistakes compounded by DKMS set it and forget it operations.

A simple failed sanity check with a clear warning about kernel compatibility would effectively stop these issues being raised.

**Comment by olegon-ru** [comment]
@mskarbek , we opened https://github.com/openzfs/zfs/wiki/Fedora and you can see
"Fedora Releases: 30, 31" there, so supposed to support Fedora 31. It's no mention of kernel version on this page. I'm not developer, and not to read all META files before upgrade system, but like ZoL very much. Please, don't flood this topic. We hope to see any ZoL developers answers, and hope to see Fedora support in ZoL again...

**Comment by glbyers** [comment]
> It always baffles me, with every issue like this, what is it so hard to understand in the phrase "Compatible with 2.6.32 - 5.4 Linux kernels" on the release page/in the META file?

Yes. What a load of elitist bullcrap. As pointed out, ZoL install page shows Fedora 31 supported through DKMS without so much as a warning. If you expect all users to be running ZoL on enterprise systems, you can expect that level of due diligence (but in reality, what enterprise would run ZoL with no insurance..). For average Joe running ZoL across 12TB of WD reds for NAS storage, you can't expect them to read the META file of every DKMS configured module prior to doing their regular "dnf upgrade --refresh". That's just nonsense, and it's an embarrassing situation for the developers. Don't deflect. The "not my fault" runs thick.

**Comment by daniel-ayers** [comment]
> It always baffles me, with every issue like this, what is it so hard to understand in the phrase "Compatible with 2.6.32 - 5.4 Linux kernels" on the release page/in the META file?

ZoL either supports fc31 or it doesn't.  Can't have it both ways.  Support for fc31 is claimed ...

**Comment by olegon-ru** [comment]
Excuse me, but where is next releases schedule page?

**Comment by tonyhutter** [comment]
zfs-0.8.4 was just released today with 5.6 kernel support.  Please try reinstalling the package.

**Comment by MartenBE** [comment]
It works! Thank you very much :thumbsup: !
